pkg/cloudfoundry/CloudFoundry_test.go

```diff
@@ -1,46 +1,207 @@
 package cloudfoundry
 
 import (
-	"testing"
-
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/mock"
 	"github.com/stretchr/testify/assert"
+	"testing"
 )
 
+func loginMockCleanup(m *mock.ExecMockRunner) {
+	m.ShouldFailOnCommand = map[string]error{}
+	m.StdoutReturn = map[string]string{}
+	m.Calls = []mock.ExecCall{}
+}
+
 func TestCloudFoundryLoginCheck(t *testing.T) {
-	t.Run("CF Login check: missing parameter", func(t *testing.T) {
+
+	m := &mock.ExecMockRunner{}
+
+	defer func() {
+		c = &command.Command{}
+	}()
+	c = m
```

Comment:
More a general remark. 
As we now sometimes pass the ExecRunner as parameter and sometimes do it like this, I think it would be good in general if we align on one approach to have a more consistent coding.

Reply:
As you can see I prefer having such kind of helpers as members.

* pro
  * smaller signature, this means the real stuff which needs handed over to functions can be seen better with less noise. Beside that there is also a code climate check claiming about too much parameters in the signatures.
* cons
  * no parallel test execution, at least not that easy.

More in the meta level: hard to align on everything in case several people are working on the same coding. So some extend we need to be tolerant to other approaches. IMO it is not necessarily a benefit to align everything. Sometimes some creative power is also good. Beside that: also in real life we are contained in a pluralistic society when in a lot of dimensions several approaches coexists. [very meta]: Would be nice to discuss how the values of a society influences coding.